js/modules/screen-reader.src.js

```diff
@@ -772,6 +773,60 @@ H.Chart.prototype.addScreenReaderRegion = function (id, tableId) {
 };
 
 
+// Add ARIA to legend
+addEvent(H.Legend, 'afterRender', function () {
+    var group = this.group,
+        items = this.allItems,
+        chart = this.chart;
+    if (group && items && items.length) {
+        group.attr({
+            role: 'region',
+            'aria-label': chart.langFormat('accessibility.legendLabel')
+        });
+
+        if (this.box) {
+            this.box.attr('aria-hidden', 'true');
+        }
+
+        items.forEach(function (item) {
+            var itemGroup = item.legendGroup,
+                text = item.legendItem,
+                visible = item.visible,
+                label = chart.langFormat(
+                    'accessibility.legendItem',
+                    {
+                        chart: chart,
+                        itemName: stripTags(item.name)
+                    }
+                );
+            if (itemGroup && text) {
+                itemGroup.attr({
+                    role: 'button',
+                    'aria-pressed': visible ? 'false' : 'true'
+                });
+                if (label) {
+                    itemGroup.attr('aria-label', label);
+                }
+                text.attr('aria-hidden', 'false');
+            }
+        });
+    }
+});
+
+
+// Handle show/hide series
```

Comment:
Legend items can also belong to Points, like the case is with the pie series. We have a legend event, `afterColorizeItem`, maybe it's more appropriate to use that? One of its parameters is `visible`.

Reply:
Added